pkg/iac/adapters/terraform/aws/iam/roles_test.go

```diff
@@ -377,3 +379,198 @@ data "aws_partition" "current" {}
+  policy_arn = aws_iam_policy.administrative_policy.arn
+}
+`
+
+	modules := tftestutil.CreateModulesFromSource(t, source, ".tf")
+	attachments := modules.GetResourcesByType("aws_iam_role_policy_attachment")
+
+	// Debug output - let's see what's actually happening
+	fmt.Printf("DEBUG: Total resources found: %d\n", len(attachments))
+	for i, attachment := range attachments {
+		fmt.Printf("DEBUG: Attachment %d: %s\n", i, attachment.Reference().String())
+		fmt.Printf("  - FullName: %s\n", attachment.FullName())
+		fmt.Printf("  - TypeLabel: %s\n", attachment.TypeLabel())
+		fmt.Printf("  - NameLabel: %s\n", attachment.NameLabel())
+		if key := attachment.Reference().RawKey(); !key.IsNull() && key.IsKnown() {
+			fmt.Printf("  - Key: %s (%s)\n", key.GoString(), key.Type().GoString())
+		}
+	}
+
+	// Also check all blocks to see what's available
+	allBlocks := modules.GetBlocks()
+	fmt.Printf("\nDEBUG: All blocks:\n")
+	for i, block := range allBlocks {
+		if strings.Contains(block.Reference().String(), "platform") {
+			fmt.Printf("  Block %d: %s (%s)\n", i, block.Reference().String(), block.Type())
+		}
+	}
+
+	var attachmentRefs []string
+	for _, a := range attachments {
+		attachmentRefs = append(attachmentRefs, a.Reference().String())
+	}
+
+	// Expected attachment keys (they will include the map key)
+	expectedCount := 2
+
+	sort.Strings(attachmentRefs)
+
+	if len(attachments) != expectedCount {
+		t.Fatalf("expected %d policy attachments, got %d: %v", expectedCount, len(attachments), attachmentRefs)
+	}
+
+	// Additional sanity check: ensure both map keys appear once
+	hasLambda := false
+	hasEcs := false
+	for _, ref := range attachmentRefs {
+		if strings.Contains(ref, "lambda") {
+			hasLambda = true
+		}
+		if strings.Contains(ref, "ecs-tasks") {
+			hasEcs = true
+		}
+	}
+	if !hasLambda || !hasEcs {
+		t.Fatalf("expected attachment refs to include both lambda and ecs-tasks keys, got %v", attachmentRefs)
+	}
+}
+
+func Test_simpleForEachReference(t *testing.T) {
+	// Test 1: Simple for_each with direct reference - this should work
+	source1 := `
+locals {
+  roles = {
+    lambda    = "lambda.amazonaws.com"
+    ecs-tasks = "ecs-tasks.amazonaws.com"
+  }
+}
+
+resource "aws_iam_role" "platform_role" {
+  for_each = local.roles
+  name     = "platform-${each.key}"
+}
+
+resource "aws_iam_role_policy_attachment" "test" {
+  for_each = aws_iam_role.platform_role
+  role     = each.value.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonS3FullAccess"
+}`
+
+	modules1 := tftestutil.CreateModulesFromSource(t, source1, ".tf")
+	attachments1 := modules1.GetResourcesByType("aws_iam_role_policy_attachment")
+
+	fmt.Printf("Test 1 - Direct reference - Attachments found: %d\n", len(attachments1))
+	for i, attachment := range attachments1 {
+		fmt.Printf("  %d: %s\n", i, attachment.Reference().String())
+		if key := attachment.Reference().RawKey(); !key.IsNull() && key.IsKnown() {
+			fmt.Printf("     Key: %s\n", key.GoString())
+		}
+	}
```

Comment:
You can use Go subtests to add multiple tests here.

Reply:
ok, let me know if the refactor looks good with subtests now